client_test/integration_test.go

```diff
@@ -0,0 +1,178 @@
+package client_test
+
+import (
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/client"
+	"github.com/statechannels/go-nitro/client/query"
+	"github.com/statechannels/go-nitro/internal/testactors"
+	td "github.com/statechannels/go-nitro/internal/testdata"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+var simpleCase = TestCase{
+	Description:    "Simple test",
+	Chain:          MockChain,
+	MessageService: TestMessageService,
+	NumOfChannels:  1,
+	MessageDelay:   0,
+	LogName:        "simple_integration_run.log",
+	NumOfHops:      1,
+	NumOfPayments:  1,
+	Participants: []TestParticipant{
+		{StoreType: MemStore, Name: testactors.AliceName},
+		{StoreType: MemStore, Name: testactors.BobName},
+		{StoreType: MemStore, Name: testactors.IreneName},
+	},
+}
+
+var complexCase = TestCase{
+	Description:    "Complex test",
+	Chain:          SimulatedChain,
+	MessageService: P2PMessageService,
+	NumOfChannels:  5,
+	MessageDelay:   0,
+	LogName:        "complex_integration_run.log",
+	NumOfHops:      2,
+	NumOfPayments:  5,
+	Participants: []TestParticipant{
+		{StoreType: DurableStore, Name: testactors.AliceName},
+		{StoreType: DurableStore, Name: testactors.BobName},
+		{StoreType: DurableStore, Name: testactors.IreneName},
+		{StoreType: DurableStore, Name: testactors.BrianName},
+	},
+}
+
+var cases = []TestCase{simpleCase, complexCase}
+
+func TestClientIntegration(t *testing.T) {
+	// Clean up all the test data we create at the end of the test
+	defer os.RemoveAll(STORE_TEST_DATA_FOLDER)
+
+	for _, tc := range cases {
```

Comment:
While we have a small number of test cases, it might be nice to spit every test case into a separate function so that we can run the test case individually. It would make the debugging experience nicer.

Reply:
Updated in https://github.com/statechannels/go-nitro/pull/1193/commits/ab2ca01266ba8a86d0687dcadc6a72418111dc0a